ipa-core/src/net/mod.rs

```diff
@@ -10,3 +17,71 @@ pub use client::{ClientIdentity, MpcHelperClient};
 pub use error::Error;
 pub use server::{MpcHelperServer, TracingSpanMaker};
 pub use transport::{HttpShardTransport, HttpTransport};
+
+/// Reads certificates and a private key from the corresponding bytes
+///
+/// # Errors
+/// If no private key or certificates are found and if there are any issues with
+/// their format.
+pub fn parse_certificate_and_private_key_bytes(
+    cert_bytes: &[u8],
+    private_key_bytes: &[u8],
+) -> Result<(Vec<CertificateDer<'static>>, PrivateKeyDer<'static>), io::Error> {
+    let mut certs_reader = BufReader::new(Cursor::new(cert_bytes));
+    let mut private_key_reader = BufReader::new(Cursor::new(private_key_bytes));
+    parse_certificate_and_private_key(&mut certs_reader, &mut private_key_reader)
+}
+
+/// Reads certificates and a private key from the corresponding buffered inputs
+///
+/// # Errors
+/// If no private key or certificates are found and if there are any issues with
+/// their format.
+pub fn parse_certificate_and_private_key(
```

Comment:
You're correct this function is currently only used above but I don't see a lot of value inlining this function since we call it just a few times. Also, in the future I do imagine we might call this function from other functions.